### PR TITLE
Food decomposition can mold stuff into different items, and not just moldy messes.

### LINKED ISSUE #61188: ants have outevolved chickens
## Reproduction:
image
laid eggs shouldnt decay, its almost impossible to have a chicken coop now since it just turns into an ant minefield

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Originally the component only allowed food to decompose into the generic moldy mess. This PR changes the component so that what item the component decomposes something into can be changed per attached object. I've added a few examples; bread types > moldy bread, pizza slice types > moldy pizza slices, dead mice > moldy dead mice (they need a sprite still, I'll get to that later. Kinda catbrained rn.)

**edit:**
Makes sure that none of this moldy food touches cytology unnecessarily. 

Rotten eggs have been added (decomposed eggs), along with rotten boiled eggs. The only way to determine if an egg is good or not requires using a container of water on the egg. If it floats, it's rotten! They also do not produce ants when they decompose, so chickens no longer get eaten alive by swarms of ants for producing all these eggs. 

fixes https://github.com/tgstation/tgstation/issues/61188

## Why It's Good For The Game

We already have some moldy items, and the inability to acquire some items as a result of this component has been annoying me. Especially given that it wasn't very modifiable and hardcoded only to produce moldy messes, which seems like a lost opportunity. 

My core reasoning for making this pr was really the fact that I wasn't ever able to find mice in current rounds since the change (impacting cargo exports, recipes and the maintenance sect favour, as well as producing organic slurry), but upon looking into the component I felt I could maybe do something more than just address one issue. 

I hope it's the right solution to the problem.

## Changelog
:cl:
code: Added an argument for the decomposition component that allows for specifying what kind of item something decomposes into.
fix: Space ants no longer prey upon chickens and their produce.
add: Includes some examples from existing food and a new kind of food. Moldy bread, moldy pizza slices, rotten eggs and moldy dead mice.
/:cl:

### CODE REVIEW

**Comment by Krysonism** [comment]
>pizza slice types > moldy pizza slices

This part will break cytology balance due to the cell sample these spawn with.

**Comment by necromanceranne** [comment]
I don't mean to be negative, but...is cytology even being actively worked on? Aren't moldy messes and moldy pizza almost equivalent to each other, they seem to apply an equivalent version of the swabbable element and you can make a _lot_ of moldy messes right now.

**Comment by necromanceranne** [comment]
fun fact: after posting this pr i discovered bread turning to the darkside in my cupboard. The Illumimoldi are showing their displeasure.

**Comment by AMonkeyThatCodes** [comment]
lol just make a type of moldy pizza without the cytology stuff anne

**Comment by Krysonism** [comment]
>is cytology even being actively worked 

Yes it is.

>they seem to apply an equivalent version of the swabbable element 

Neither of them should apply that version of the swabbable element. 

It goes against the stated main goals of cytology; having to go out in the world and interact with the round. If you could just sit in your autism fort with donk pockets on the floor waiting for them to mold to collect the perfect sample, it subverts that design.

**Comment by necromanceranne** [comment]
Well sure I guess I'll fix that problem then. Did you bring that up in the pr that added this component?

**Comment by Krysonism** [comment]
Thank you for addressing my concern, I think the PR is a good idea. 

>I really do feel this is a problem that needed to be a addressed when that PR was up.

I would have pointed it out if I was aware of it, the PR was huge and that part was not listed In the changelog so I didn't catch it.

**Comment by Ghommie** [comment]
> This part will break cytology balance due to the cell sample these spawn with.

What is vatgrowing even good for outside of a couple mobs with cool sprites? Xenobiology, chemistry, botany, cargo (with enough money) and that guy with cheese wheels and mice can all spawn critters better and faster than it. In one year, I have yet to see a vat beast or jelly cube mob in game. I have seen clown mobs more often, but they're likely being spawned by admins.

> It goes against the stated main goals of cytology; having to go out in the world and interact with the round. If you could just sit in your autism fort with donk pockets on the floor waiting for them to mold to collect the perfect sample, it subverts that design.

Just like for moldy ones, getting fresh pizza requires the scientist player to get out of the lab (or get lucky with silver slimes later on but it'll be more and more difficult as new food types are added). Whether they order it from cargo, ask the chef or make it yourself, you still have to go out and interact with the outside world. Such argument doesn't stand....

...Furthermore, the base moldy food item already has that swabbable element:
https://github.com/tgstation/tgstation/blob/a6bd69118bd2f71c77c0ba2ee824ee3ad121231b/code/game/objects/items/food/misc.dm#L204-L206

Your feature has been "broken" for about 1 month and a hal, yet you only care about it now? It doesn't honestly look like it's being actively worked on to me. But please don't let that think I'm verbally bashing you. This is merely fifty cents from someone who has enjoyed doing the aforementioned mob spawning jobs and gimmicks in the past but hasn't found interest in cytology yet.

The element name doesn't match the file name by the by.

**Comment by tralezab** [comment]
hahhahaha ghommie jesus christ haha

DAMN.

i must also say cytology blocking "ALL MOLDY FOOD EVER" is kind of a ridiculous design restriction

**Comment by AMonkeyThatCodes** [comment]
its more that allowing a low-effort way to make infinite of a maintenace spawn item is cringy. if none of you can comprehend that from the simple post then thats just unfortunate.

**Comment by Ghommie** [comment]
> 
> hahhahaha ghommie jesus christ haha

Yea, I may have wasted a bit too many neurons on that comment out of boredom.

>
> 
> its more that allowing a low-effort way to make infinite of a maintenace spawn item is cringy. if none of you can comprehend that from his simple post then thats just unfortunate.

Then move it away from maintenance. Scientist don't have the access anyway and the tunnels are trashy enough already.

**Comment by AMonkeyThatCodes** [comment]
shit take

**Comment by necromanceranne** [comment]
fellas, while I appreciate the discussion on moldy food and cytology here, I feel like the broad gameplay implication discussions needs to be discussed someplace else. I'm already going to do the work to make sure this side steps cytology and course corrects this as requested. 

Also, I'll get to fixing this pr in a bit, I've got a lot on my plate rn irl so hopefully I can fix the conflicts tonight.

File: code/game/objects/items/food/misc.dm [hunk 1]
```diff
@@ -201,7 +201,11 @@
 	desc = "A rancid, disgusting culture of mold and ants. Somewhere under there, at <i>some point,</i> there was food."
 	food_reagents = list(/datum/reagent/consumable/mold = 30)
 
-/obj/item/food/badrecipe/moldy/Initialize()
+/obj/item/food/badrecipe/moldy/bacteria
+	name = "bacterua rich moldy mess"
```
Comment (Wallemations) [inline code/game/objects/items/food/misc.dm:205]: you misspelled bacteria here

**Comment by Wallemations** [comment]
Hell yea, nice.
I hadn't thought about the different types of moldy foods that already existed in the game. Good work on this.
One thing though, if you're going to add that bacteria version of moldy bread, should the vending machine in permabrig sell moldy bread with or without bacteria?

File: code/game/objects/items/food/misc.dm (hunk 1 above)
Comment (necromanceranne) [reply code/game/objects/items/food/misc.dm:205]: Nice catch

**Comment by necromanceranne** [comment]
Given current discussion, I'm going to go with the safe route and say without.

**Comment by Wallemations** [comment]
Apparently eggs can hatch into chickens, if this is true, do you think you could figure out a way to make eggs laid by chickens have `var/decomp_req_handle = TRUE`? If I'm not mistaken, the code for that is in `/datum/component/egg_layer`. 
If eggs do not naturally spawn into chickens, then you can disregard this lmao.

This was apparently part of the problem that was brought up in #61188.

**Comment by necromanceranne** [comment]
I think since apparently decomp_req_handle is checked on the component's initialization that it would be easier to just make eggs have decomp_req_handle by default rather than fiddle with the component when the egg randomly is determined to lay an egg (which likely comes after initialization). I don't think it'll change much, your mountain of eggs are just immune to decomp until you pick them up.

**Comment by Wallemations** [comment]
> I think since apparently decomp_req_handle is checked on the component's initialization that it would be easier to just make eggs have decomp_req_handle by default rather than fiddle with the component when the egg randomly is determined to lay an egg (which likely comes after initialization). I don't think it'll change much, your mountain of laid eggs are just immune to decomp until you pick them up.

That's fair, thanks for fixin' that either way.

**Pull request merged**